processor/maskprocessor/README.md

```diff
@@ -0,0 +1,79 @@
+# Mask Processor
+This processor is used to detect and mask sensitive data.
+
+## Supported pipelines
+- Logs
+- Metrics
+- Traces
+
+## How It Works
+1. This processor traverses the attribute and resource fields of incoming telemetry. For log-based telemetry, it will also traverse the body.
+2. If a field matches a defined regex rule, the matching value is replaced with `[masked_value]`, where `value` is the name of the  rule. For instance, a rule that masks email addresses would result in `[masked_email]`.
+3. If a field is to be excluded from masking, it can be specified in the processor's `exclude` field. By default, the processor will mask all fields.
```

Comment:
Might be good to call out that this only works for string values (e.g. a social security number wouldn't be masked if it was encoded as an int for some reason)